### PR TITLE
Add NonNullCollection to Primitives

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Copying NonNullCollection from the designer and use as the base for GridEntryCollection to type the collection and add null safety.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/5499)

### CODE REVIEW

**Comment by JeremyKuhne** [comment]
x86 runs failed with #4347

**Review by Tanya-Solyanik — DISMISSED** [review]
:shipit:

File: src/System.Windows.Forms/src/System/Windows/Forms/GridItemCollection.cs [hunk 1]
```diff
@@ -12,20 +12,20 @@ namespace System.Windows.Forms
     public class GridItemCollection : ICollection
     {
 #pragma warning disable IDE1006 // Naming Styles - this is public API
-        public static GridItemCollection Empty = new(Array.Empty<GridItem>());
+        public static GridItemCollection Empty = new(null);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/GridItemCollection.cs:15]:
```suggestion
        public static GridItemCollection Empty = new(entries: null);
```

**Review by RussKie — APPROVED** [review]
👍

**Pull request merged**